docs/api.rst

```diff
@@ -3,8 +3,72 @@ Python API
 
 Pipeline
 --------
+In DALI, any data processing task has a central object called pipeline. Pipeline object is an
+instance of :class:`nvidia.dali.pipeline.Pipeline` or a derived class. Pipeline encapsulates the
+data processing graph and the execution engine.
+
+There are two ways to define a DALI pipelines:
+
+#. by inheriting from Pipeline class and overriding :meth:`nvidia.dali.pipeline.Pipeline.define_graph`
+#. by instantiating `Pipeline` directly, building the graph and setting the pipeline
```

Comment:
I would indicate this is experimental or we want to commit to fully support it from now?

Reply:
I'd say that this is not really experimental - it's dirt-cheap (one-liner) to maintain for us and frees the user from the need to inherit the pipeline (tedious) or overwriting define_graph (ugly).